Source/Core/Duality/Utility/ReflectionHelper.cs

```diff
@@ -121,7 +121,10 @@ private static class StaticLookup<T>
 				}
 
 				// Mind the result for later. Don't do this twice.
-				customMemberAttribCache[member] = result;
+				lock (customMemberAttribCache)
```

Comment:
It's a best practice in C# to avoid locking on anything other than a `private object fooLock = new object();` created specifically for that purpose, though as long as the collections we're dealing with here are private themselves, I could live with it. Mentioning it, but I'll leave this one to you.